### ISSUE TITLE
Sorting by another column inverts the current order

### ISSUE THREAD

**Issue by jesus2099** [OpenUserJs/OpenUserJS.org#803]
Let’s say you have clicked **Installed** to sort by descending download times.
Now you click the next column (**Ratings** I guess), it will now sort **ascending** instead of **descending**.
It should keep the last selected order, IMO. :)

**Comment by sizzlemctwizzle** [comment]
Using the default order for the newly selected column makes sense to me.
Can you give me examples of other applications that exhibit the behavior
you want?

**Comment by jesus2099** [comment]
> Using the default order for the newly selected column makes sense to me.

Yes but it does not. Let me be more clear on what is right and what seems wrong to me.  :)

https://openuserjs.org shows scripts sorted by unknown column, unknown direction. From here : 

1. Clicking [Name](https://openuserjs.org/?orderBy=name&orderDir=asc) is **ascending** (good default)
  1. Clicking [Name](https://openuserjs.org/?orderBy=name&orderDir=desc) is **descending** (good opposite)
    1. Clicking [Name](https://openuserjs.org/?orderBy=name&orderDir=asc) is **ascending** (good opposite) → goto 1.
    2. Clicking [Installs](https://openuserjs.org/?orderBy=installs&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
    3. Clicking [Rating](https://openuserjs.org/?orderBy=rating&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
    4. Clicking [Last Updated](https://openuserjs.org/?orderBy=updated&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
  2. Clicking [Installs](https://openuserjs.org/?orderBy=installs&orderDir=desc) is **descending** (good by luck) → goto 2.
  3. Clicking [Rating](https://openuserjs.org/?orderBy=rating&orderDir=desc) is **descending** (good by luck) → goto 3.
  4. Clicking [Last Updated](https://openuserjs.org/?orderBy=updated&orderDir=desc) is **descending** (good by luck) → goto 4.
2. Clicking [Installs](https://openuserjs.org/?orderBy=installs&orderDir=desc) is **descending** (good default)
  1. Clicking [Name](https://openuserjs.org/?orderBy=name&orderDir=asc) is **ascending** (good by luck) → goto 1.
  2. Clicking [Installs](https://openuserjs.org/?orderBy=installs&orderDir=asc) is **ascending** (good opposite)
  3. Clicking [Rating](https://openuserjs.org/?orderBy=rating&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
  4. Clicking [Last Updated](https://openuserjs.org/?orderBy=updated&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
3. Clicking [Rating](https://openuserjs.org/?orderBy=rating&orderDir=desc) is **descending** (good default)
  1. Clicking [Name](https://openuserjs.org/?orderBy=name&orderDir=asc) is **ascending** (good by luck) → goto 1.
  2. Clicking [Installs](https://openuserjs.org/?orderBy=installs&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
  3. Clicking [Rating](https://openuserjs.org/?orderBy=rating&orderDir=asc) is **ascending** (good opposite)
  4. Clicking [Last Updated](https://openuserjs.org/?orderBy=updated&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
4. Clicking [Last Updated](https://openuserjs.org/?orderBy=updated&orderDir=desc) is **descending** (good default)
  1. Clicking [Name](https://openuserjs.org/?orderBy=name&orderDir=asc) is **ascending** (good by luck) → goto 1.
  2. Clicking [Installs](https://openuserjs.org/?orderBy=installs&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
  3. Clicking [Rating](https://openuserjs.org/?orderBy=rating&orderDir=asc) is **ascending** (**NO GOOD** :red_circle:)
  4. Clicking [Last Updated](https://openuserjs.org/?orderBy=updated&orderDir=asc) is **ascending** (good opposite)

I could write all the cases but It’s becoming more and more confusing, I think now you can see my point…

Each opposite click is OK, but the problem is that the default is not used on other columns, it’s always using the same as the opposite click.

**Comment by Martii** [comment]
> https://openuserjs.org shows scripts sorted by unknown column, unknown direction.

That would be https://openuserjs.org/?orderBy=rating&orderDir=desc I believe

> Clicking Name is ascending (good default) 

True

> Clicking Name is descending (good opposite) 

True

> Clicking Name is ascending (good opposite) → goto 1.

True... went to 1. since you didn't say go back to b. I did it anyhow and it generates:

> Clicking Installs is ascending (NO GOOD :red_circle:)

**FALSE**... the URI is https://openuserjs.org/?orderBy=installs&orderDir=desc and Quack Toolsammlung is the top installed on the site and at the top of the list here as expected.

Try to reduce what you are experiencing down. Having that many test cases is too much.

That routine is handled [currently here with the defaults](/OpenUserJs/OpenUserJS.org/blob/73ebe0263ffb218f27d8ade2fad7957ae791b2bf/controllers/index.js#L44-L47) and [https://github.com/OpenUserJs/OpenUserJS.org/blob/83096c59cfa4091d10b1d3ab3de0434b65acddf6/libs/templateHelpers.js#L172-L183](currently here with the toggling)... if you can find something that is more agreeable to what you are looking for please post the logic behind it... however from here your use cases fail in the first test with a false report of NO GOOD.

**Comment by Martii** [comment]
So to elaborate on the current logic on a very reduced test case...

1. Start at https://openuserjs.org/ ... all initial defaults are in place since there aren't any QSPs
2. Clicking "Name" pushes you into using the QSPs... where the toggle function will invert what you are currently sorting by... this is intended behavior.

Now if you start at a QSPd URI of:

1. https://openuserjs.org/?orderBy=rating&orderDir=desc ... the initial defaults are not applied because you are using the QSPs... you'll notice all the `orderDir`ections are the same.
2. So the toggle function will invert your sort order... again intended behavior.

If you can find some code logic that changes it to what you are expecting I'll try it out... but as far as I can tell this is intended behavior.

**Comment by Martii** [comment]
Another example of this is:

1. Start out by using https://openuserjs.org/?orderBy=updated and that uses the initial default `orderDir`ection for all table headers... as expected since the toggle routine isn't requested at this point.

**Comment by jesus2099** [comment]
Ok my explanation was even less understandable than first post, sorry. :)
Let's say I go to [my page](https://openuserjs.org/users/jesus2099/scripts) where I want to see latest updated.
I click **Last Updated**.
Now I want to see most downloaded, so click **Installs**.
Here is the problem, now I see the **Least installed**.
Same problem if I had wanted the best noted.
It keeps going back and forth between best and worst and it is unusual user experience, no?

**Comment by Martii** [comment]
I'm tinkering with it now... I'll see if I can get it your style... but I may have to do this in a User preference to accommodate sizzle and the intended defaults... still running through some permutations... basically what you are looking for is the table headers not to use the `orderDir` with the exception of if you are on the `orderBy`... give me some time to ponder on it.

**Comment by Martii** [comment]
GH is also not cooperating very well in dev environment so difficult to test at this time.

**Comment by jesus2099** [comment]
Yes yes please take your time to think it... Compare with what you usually expect or have elsewhere.
I may be mistaking so it is good when more people think it...

**Comment by Martii** [comment]
@sizzlemctwizzle 
The commit above might be feasible... please give it a whirl under the stated conditions for now on dev and local pro and see if this is something that you want to implement as a default or as an option. To me the coding itself makes more sense this way although I think `orderDir()` needs to be combined into this somehow.
